src/connector/src/sink/encoder/json.rs

```diff
@@ -289,11 +295,27 @@ fn datum_to_json_object(
         (DataType::Interval, ScalarRefImpl::Interval(v)) => {
             json!(v.as_iso_8601())
         }
-        (DataType::Jsonb, ScalarRefImpl::Jsonb(jsonb_ref)) => match &config.custom_json_type {
-            CustomJsonType::Es | CustomJsonType::StarRocks => JsonbVal::from(jsonb_ref).take(),
-            CustomJsonType::Doris(_) | CustomJsonType::None => {
+
+        (DataType::Jsonb, ScalarRefImpl::Jsonb(jsonb_ref)) => match config.custom_json_type {
+            CustomJsonType::Es | CustomJsonType::StarRocks => {
+                matches!(config.encode_jsonb_mode, EncodeJsonbMode::Custom);
+                JsonbVal::from(jsonb_ref).take()
+            }
+            CustomJsonType::Doris(_) => {
+                matches!(config.encode_jsonb_mode, EncodeJsonbMode::Custom);
```

Comment:
It looks like ES & StarRocks can just use `dynamic` and Doris uses `string`. But I am okay with keeping them for now if you prefer not to take the risk.

Reply:
hmm I guess we can do that and remove `EncodeJsonbMode::Custom`